GVFS/GVFS/CommandLine/CloneVerb.cs

```diff
@@ -567,6 +567,27 @@ private void CheckNotInsideExistingRepo(string normalizedEnlistmentRootPath)
             }
 
             GitProcess.Result forceCheckoutResult = git.ForceCheckout(branch);
+            if (forceCheckoutResult.HasErrors)
+            {
+                // It is possible to have the above TryDownloadCommit() fail because we
+                // already have the commit and root tree we intend to check out, but
+                // don't have a tree further down the working directory. If we fail
+                // checkout here, it may be due to not having these trees and the
+                // read-object hook is not available yet. Force downloading the commit
+                // again and retry the checkout.
+
+                if (!this.ForceTryDownloadCommit(

```

Comment:
How much work would it be to write a functional test for this scenario?

Reply:
I've thought about how to set this up, and it is really difficult to do. I don't think the severity of this issue is worth engineering time on the test.

Reply:
What is the difficulty? We already have some tests that can work with multiple repos, so at first glance, it seems like you could just clone once at one branch, touch some other branch's root tree object, then clone again at that second branch.

Reply:
The problem is that you need to somehow guarantee a missing tree. This means we need to disable the background prefetch for the first enlistment in order to ensure we don't get the objects.

Reply:
Would turning on unattended mode for this test achieve that?

Reply:
Or alternatively, after you unmount the first repo, delete any packfiles from its cache.

Reply:
Actually, this should be a non-issue. Background prefetch is already disabled if there is no cache server configured, and our functional tests do not use a cache server.

Reply:
It took some trial-and-error, but deleting the packfiles, downloading the commit and root tree as loose, then re-cloning worked. I ran the full functional test suite to verify and inspected the logs to ensure it did everything as expected.